lib/index.js

```diff
@@ -263,42 +315,152 @@ api.get = async ({actor, account, email, type} = {}) => {
   } else {
     query['account.email'] = email;
   }
+
+  const tokenId = 'meta.bedrock-authn-token.tokens.nonce.id';
+  if(id) {
+    query[tokenId] = id;
+  }
+
+  const record = await database.collections.account.findOne(
+    query, {projection: {_id: 0, [typeKey]: 1}});
+
+  if(!record) {
+    throw new BedrockError(
+      'Authentication token not found.',
+      'NotFoundError', {
+        httpStatusCode: 404,
+        public: true
+      });
+  }
+
+  let result = record.meta['bedrock-authn-token'].tokens[type];
+  let rValue;
+  if(type === 'nonce') {
+    if(id) {
+      result = result.filter(token => token.id === id);
+    }
+    rValue = result[0];
+  } else {
+    rValue = result;
+  }
+  // filter expired token.
+  if(filterExpiredTokens) {
+    const now = Date.now();
+    if(rValue.expires < now) {
+      // throw NotFoundError
+      throw new BedrockError(
+        'Authentication token has expired.',
+        'NotFoundError', {
+          httpStatusCode: 404,
+          public: true
+        });
+    }
+  }
+  return rValue;
+};
+
+/**
+ * Gets tokens for an account.
+ *
+ * @param {object} options - The options to use.
+ * @param {object} options.actor - The actor or capabilities to perform
+ *   the action.
+ * @param {string} [options.account] - The ID of the account.
+ * @param {string} [options.email] - The email of the account.
+ * @param {string} options.type - The type of tokens to get.
+ * @param {boolean} [options.filteredExpiredTokens] - Set to `true` to drop
+ *   expired tokens and `false` to get all tokens including expired tokens.
+ *
+ * @return {Promise<array>} - A Promise that resolves to tokens.
+ */
+api.getAll = async ({
+  actor, account, email, type, filterExpiredTokens = true
+} = {}) => {
+  assert.optionalString(account, 'account');
+  assert.optionalString(email, 'email');
+  assert.optionalBool(filterExpiredTokens, 'filterExpiredTokens');
+
+  if(!(account || email)) {
+    throw new Error('Either "account" or "email" must be provided.');
+  }
+  validateTokenType(type);
+
+  await brPermissionCheck(
+    actor, PERMISSIONS.ACCOUNT_ACCESS, {resource: [account]});
+
+  // get tokens from meta
+  const typeKey = `meta.bedrock-authn-token.tokens.${type}`;
+  const query = {};
+  if(account) {
+    query.id = database.hash(account);
+  } else {
+    query['account.email'] = email;
+  }
   const record = await database.collections.account.findOne(
     query, {projection: {_id: 0, [typeKey]: 1}});
+
   if(!record) {
     throw new BedrockError('Authentication token not found.', 'NotFoundError', {
       httpStatusCode: 404,
       public: true
     });
   }
+  // check if any of the keys are undefined or null
+  let result = typeKey.split('.').reduce(function(acc, next) {
+    return (typeof acc == 'undefined' || acc === null) ? acc : acc[next];
+  }, record);
```

Comment:
Based on the comment, it seems that `.some()` should be used instead of `reduce`.

Reply:
This code may be a bit too tricky. :-)  I think perhaps https://lodash.com/docs/4.17.15#get would work here?  Something like:
```js
let result = _.get(record, typeKey, []);
```

Reply:
lodash.get is a personal favorite of mine.  If anyone decides to use it, please use the tiny package, not the full lodash kitchen sink.  https://www.npmjs.com/package/lodash.get

Reply:
Bedrock may already pull in the kitchen sink.  Maybe should use smaller packages there too.

Reply:
Updated.